contracts/contracts/lib/CrossMsgHelper.sol

```diff
@@ -131,6 +135,22 @@ library CrossMsgHelper {
         return keccak256(abi.encode(crossMsg));
     }
 
+    /// @notice Returns a deterministic hash for the given cross message. The hash remains the same accross different networks
+    /// because it doesn't include the network-specific nonce.
+    function toDeterministicHash(IpcEnvelope memory crossMsg) internal pure returns (bytes32) {
```

Comment:
I guess most of the time this is ok, but will this cause conflict if say someone sends the same message twice?

Reply:
It might collide, but it is only for logging so it should not matter.